lnrpc/walletrpc/walletkit_server.go

```diff
@@ -409,3 +416,107 @@ func (w *WalletKit) PendingSweeps(ctx context.Context,
 		PendingSweeps: rpcPendingSweeps,
 	}, nil
 }
+
+// unmarshallOutPoint converts an outpoint from its lnrpc type to its canonical
+// type.
+func unmarshallOutPoint(op *lnrpc.OutPoint) (*wire.OutPoint, error) {
+	var hash chainhash.Hash
+	switch {
+	case len(op.TxidBytes) == 0 && len(op.TxidStr) == 0:
+		fallthrough
+
+	case len(op.TxidBytes) != 0 && len(op.TxidStr) != 0:
+		return nil, fmt.Errorf("either TxidBytes or TxidStr must be " +
+			"specified, but not both")
+
+	// The hash was provided as raw bytes.
+	case len(op.TxidBytes) != 0:
+		copy(hash[:], op.TxidBytes)
+
+	// The hash was provided as a hex-encoded string.
+	case len(op.TxidStr) != 0:
+		h, err := chainhash.NewHashFromStr(op.TxidStr)
+		if err != nil {
+			return nil, err
+		}
+		hash = *h
+	}
+
+	return &wire.OutPoint{
+		Hash:  hash,
+		Index: op.OutputIndex,
+	}, nil
+}
+
+// BumpFee allows bumping the fee rate of an arbitrary input. A fee preference
+// can be expressed either as a specific fee rate or a delta of blocks in which
+// the output should be swept on-chain within. If a fee preference is not
+// explicitly specified, then an error is returned. The status of the input
+// sweep can be checked through the PendingSweeps RPC.
+func (w *WalletKit) BumpFee(ctx context.Context,
+	in *BumpFeeRequest) (*BumpFeeResponse, error) {
+
+	// Parse the outpoint from the request.
+	op, err := unmarshallOutPoint(in.Outpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct the request's fee preference.
+	satPerKw := lnwallet.SatPerKVByte(in.SatPerByte * 1000).FeePerKWeight()
+	feePreference := sweep.FeePreference{
+		ConfTarget: uint32(in.TargetConf),
+		FeeRate:    satPerKw,
+	}
+
+	// We'll retrieve the set of inputs the UtxoSweeper is currently
+	// attempting to sweep. If the input is found within it, then we can
+	// simply bump its fee, which will result in a replacement transaction
+	// (RBF) being broadcast.
+	pendingInputs, err := w.cfg.Sweeper.PendingInputs()
+	if err != nil {
+		return nil, err
+	}
+
+	if _, ok := pendingInputs[*op]; ok {
```

Comment:
Is this necessary or can you rely on `ErrNotMine` returned from `BumpFee` ?

Reply:
Done.